src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java

```diff
@@ -59,92 +68,158 @@
 
   private static final char SEPARATOR = ',';
 
-  private static final String JAR_REGEXP = "(.*)!/";
-  private static final Pattern JAR_PATTERN = Pattern.compile(JAR_REGEXP);
-
   @Override
   public Map<String, String> configure(@NotNull Module ijModule) {
+    JavaModuleClasspath moduleClasspath = new JavaModuleClasspath();
+    moduleClasspath.dependentModules().add(ijModule);
+    collectModuleClasspath(moduleClasspath, ijModule, true, false);
```

Comment:
I think we could simplify those boolean flags. 

In collectModuleClasspath if topLevel==false we do nothing inside the loop, so in this case all we do in the method is call processCompilerOutput. So instead of calling collectModuleClasspath with topLevel=false, we could directly call processCompilerOutput (probably a null check should also be moved). We can now get rid of topLevel that is now always true.

The second flag can then also be simplified in collectModuleClasspath because it is now always false.

In the end this would simplify collectModuleClasspath.

Reply:
> if topLevel==false we do nothing inside the loop

I double checked, and I disagree. We do nothing only if topLevel == false AND entry is not exported.

Reply:
OK, got it... I wanted to try to simplify things a bit, but it seems we can't